target/linux/ipq40xx/files-4.19/arch/arm/boot/dts/qcom-ipq4018-dlink-dap-2610.dts

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qcom-ipq4019-ap.dk01.1.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "D-Link DAP 2610";
+	compatible = "dlink,dap-2610";
+
+	soc {
+		edma@c080000 {
+			qcom,num_gmac = <1>;
+		};
+	};
+
+	aliases {
+		led-boot = &power;
+		led-failsafe = &power;
+		led-running = &power;
+		led-upgrade = &power;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		red {
+			label = "dap-2610:red:power";
+			gpios = <&tlmm 4 GPIO_ACTIVE_LOW>;
+		};
+
+		power: green {
+			label = "dap-2610:green:power";
+			gpios = <&tlmm 5 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&tlmm 63 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&gmac0 {
+	vlan_tag = <2 0x20>;
+};
+
+&blsp1_spi1 {
+	flash@0 {
+		compatible = "w25q128", "jedec,spi-nor";
```

Comment:
w25q128 is not needed here, "jedec,spi-nor" will auto id it.

Reply:
The included qcom-ipq4019-ap.dk01.1.dtsi contains a "compatible = mx25l25635e" for this node. So while it is not required, omitting it here will lead to a warning similar to "expected mx25..., found w25q.." when the chip is detected.  
I'm fine either way, or even removing the compatible from the included dtsi.

Reply:
Humm, you are correct. I cannot provoke this warning anymore. I will change to only "jedec,spi-nor".  
(My understanding of how nodes in device-tree are overridden is somewhat lacking. It seems my flash@0 node replaces the mx25l25635e@0 from the dtsi file, as opposed to only overriding individual values.)

Reply:
Even if that warning occurs its not a issue, its just a warning and NOR will get properly identified.
You usually add the compatible if there is some special quirk or you have a model which does not auto ID properly.